compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -1008,6 +1008,70 @@ impl<'p, 'tcx> TypeCx for RustcMatchCheckCtxt<'p, 'tcx> {
             errors::OverlappingRangeEndpoints { overlap: overlaps, range: pat_span },
         );
     }
+
+    fn lint_non_contiguous_range_endpoints(
+        &self,
+        pat: &crate::pat::DeconstructedPat<'_, Self>,
+        gap: IntRange,
+        gapped_with: &[&crate::pat::DeconstructedPat<'_, Self>],
+    ) {
+        let thir_pat = *pat.data().unwrap();
```

Comment:
Can we make this a let else instead? I fear stray `unwrap`s even if the code is correct today, as it might not be in the future.